src/components/Camera/PhotoPreview.js

```diff
@@ -84,13 +67,40 @@ const PhotoPreview = ( { photos, setPhotos }: Props ): Node => {
   );
 
   return (
-    <FlatList
-      data={photos}
-      contentContainerStyle={viewStyles.photoContainer}
-      renderItem={renderSmallPhoto}
-      horizontal
-      ListEmptyComponent={emptyDescription}
-    />
+    <>
+      <Portal>
+        <Dialog visible={deleteDialogVisible} onDismiss={hideDialog}>
+          <Dialog.Content>
+            <Paragraph>{t( "Are-you-sure" )}</Paragraph>
+          </Dialog.Content>
+          <Dialog.Actions>
+            <Button onPress={hideDialog} style={viewStyles.cancelButton}>
+              {t( "Cancel" )}
+            </Button>
+            <Button onPress={deletePhoto} style={viewStyles.confirmButton}>
+              {t( "Yes-delete-photo" )}
+            </Button>
+          </Dialog.Actions>
+        </Dialog>
+      </Portal>
+      <Portal>
+        <Modal visible={mediaViewerVisible} onDismiss={hideModal} contentContainerStyle={viewStyles.container}>
```

Comment:
Yes, my reason for setting this up as a modal instead of a react-navigation screen was because I wanted a way to pass a callback.

Is there a way you see to do this without needing callbacks at all? I don't love having the callbacks *or* using a paper modal here, but I couldn't think of another way to rerender the photos in the StandardCamera and ObsEdit after deletion without callbacks. I guess this is another place where having a more global state -- like provider/context -- could help us show users the photos they're expecting to see without needing to pass callbacks around.

Reply:
One way to do it might be StandardCamera and ObsEdit just reload when you navigate back to them. E.g. StandardCamera could take an optional array of photo UUIDs as a navigation param, so when you enter the MediaViewer from StandardCamera and then finish, you just navigate back to StandardCamera and pass the remaining photo UUIDs as nav params. StandardCamera doesn't really have any other state to maintain.

ObsEdit is a bit trickier b/c you'd need to save a *draft* state (maybe in Realm) if you want the user to be able to undo any changes they made when they return from the MediaViewer. That might be out of scope here, but it's probably something we'll want to do eventually to deal with situations where the app gets shut down in the background while someone's in the middle of editing their observation. That's going to require writing state to disk, so provider/context might not be the best approach.

Reply:
This approach would mean we would also need to keep track of navigation and which screen a user navigated from previously, right? I don't think there's a way to pass navigation params via `navigation.goBack( )`, so we would probably need to pass a `lastScreen` param to the MediaViewer to determine whether to navigate back to StandardCamera or ObsEdit.

I'm noticing we might need to do the same thing for ObsEdit. Right now, when a user enters the MediaViewer from ObsEdit, deletes a photo, goes back to the StandardCamera, the StandardCamera will show all the photos, and only ObsEdit will reflect the deleted photos. We can pass an array of photo uris back to StandardCamera, but that means we would need to track which screen a user came from before they landed on ObsEdit and navigate accordingly.

Reply:
In theory, StandardCamera will hold on to its `photoUris` as a part of state, so no matter what screen you were on previously, when you return to it we could perform a check to make sure all those URIs actually correspond to real photos. If they no longer exist b/c they were deleted in MediaViewer or elsewhere, they could just be removed from state.